### PR TITLE
[BAEL-4281] Examples of array comparisons

### PR DESCRIPTION
**Pull request by jesperancinha**

### CODE REVIEW

File: java-collections/src/test/java/com/baeldung/DeepEqualsCompareUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    @Test
+    public void givenArray1andArray2_whenNotSameContent_thenNotDeepEquals() {
+        final Plane[][] planes1 = new Plane[][] { new Plane[] { new Plane("Plane 1", "A320") },
+            new Plane[] { new Plane("Plane 2", "B738") } };
+        final Plane[][] planes2 = new Plane[][] { new Plane[] { new Plane("Plane 2", "B738") },
+            new Plane[] { new Plane("Plane 1", "A320") } };
+
+        boolean result = Arrays.deepEquals(planes1, planes2);
+        assertFalse("Result is true", result);
```
Comment (KevinGilmore) [inline java-collections/src/test/java/com/baeldung/DeepEqualsCompareUnitTest.java:32]: Again, no need to create a variable -- put the expression in the assertion

File: java-collections/src/test/java/com/baeldung/EqualsCompareUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+public class EqualsCompareUnitTest {
+
+    @Test
+    public void givenArray1andArray2_whenSameContent_thenEquals() {
+        final String[] planes1 = new String[] { "A320", "B738", "A321", "A319", "B77W", "B737", "A333", "A332" };
+        final String[] planes2 = new String[] { "A320", "B738", "A321", "A319", "B77W", "B737", "A333", "A332" };
+
+        boolean result = Arrays.equals(planes1, planes2);
+        assertTrue("Result is not true", result);
```
Comment (KevinGilmore) [inline java-collections/src/test/java/com/baeldung/EqualsCompareUnitTest.java:18]: make this a one-liner also

File: java-collections/src/test/java/com/baeldung/EqualsCompareUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    @Test
+    public void givenArray1andArray2_whenSameContentOtherSort_thenNotEquals() {
+        final String[] planes1 = new String[] { "A320", "B738", "A321", "A319", "B77W", "B737", "A333", "A332" };
+        final String[] planes2 = new String[] { "B738", "A320", "A321", "A319", "B77W", "B737", "A333", "A332" };
+
+        boolean result = Arrays.equals(planes1, planes2);
+        assertFalse("Result is true", result);
```
Comment (KevinGilmore) [inline java-collections/src/test/java/com/baeldung/EqualsCompareUnitTest.java:27]: one-liner

File: java-collections/src/test/java/com/baeldung/OrderCompareUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+                    .compareTo(o1.getModel());
+            }
+            return o2.getName()
+                .compareTo(o1.getName());
+        };
+        Arrays.sort(planes1[0], planeComparator);
+        Arrays.sort(planes2[0], planeComparator);
+
+        boolean result = Arrays.deepEquals(planes1, planes2);
+        assertTrue("Result is false", result);
```
Comment (KevinGilmore) [inline java-collections/src/test/java/com/baeldung/OrderCompareUnitTest.java:32]: one-liner

File: java-collections/README.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+## Java Collections Examples
```
[the remark this answers is not in the document]
Comment (jesperancinha) [reply java-collections/README.md]: I have moved the code into there.

**Pull request merged**